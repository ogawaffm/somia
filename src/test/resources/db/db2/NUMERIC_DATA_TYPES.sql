DROP TABLE IF EXISTS TEST.NUMERIC_DATA_TYPES;
CREATE TABLE TEST.NUMERIC_DATA_TYPES
(
    SMALLINT_COL         SMALLINT,
    INT_COL              INT,
    INTEGER_COL          INTEGER,
    BIGINT_COL           BIGINT,

    REAL_COL             REAL,

    FLOAT_COL            FLOAT,
    FLOAT_1_COL          FLOAT(1),
    FLOAT_24_COL         FLOAT(24),
    FLOAT_25_COL         FLOAT(25),
    FLOAT_53_COL         FLOAT(53),

    DOUBLE_COL           DOUBLE,
    DOUBLE_PRECISION_COL DOUBLE PRECISION,

    DECFLOAT_COL         DECFLOAT,
    DECFLOAT_16_COL      DECFLOAT(16),
    DECFLOAT_34_COL      DECFLOAT(34),

    DEC_COL              DEC,
    DECIMAL_COL          DECIMAL,
    NUMERIC_COL          NUMERIC,
    NUM_COL              NUM,

    DEC_1_COL            DEC(1),
    DECIMAL_1_COL        DECIMAL(1),
    NUMERIC_1_COL        NUMERIC(1),
    NUM_1_COL            NUM(1),

    DEC_1_0_COL          DEC(1, 0),
    DECIMAL_1_0_COL      DECIMAL(1, 0),
    NUMERIC_1_0_COL      NUMERIC(1, 0),
    NUM_1_0_COL          NUM(1, 0),

    DEC_1_1_COL          DEC(1, 1),
    DECIMAL_1_1_COL      DECIMAL(1, 1),
    NUMERIC_1_1_COL      NUMERIC(1, 1),
    NUM_1_1_COL          NUM(1, 1),

    DEC_31_COL           DEC(31),
    DECIMAL_31_COL       DECIMAL(31),
    NUMERIC_31_COL       NUMERIC(31),
    NUM_31_COL           NUM(31),

    DEC_31_0_COL         DEC(31, 0),
    DECIMAL_31_0_COL     DECIMAL(31, 0),
    NUMERIC_31_0_COL     NUMERIC(31, 0),
    NUM_31_0_COL         NUM(31, 0),

    DEC_31_1_COL         DEC(31, 1),
    DECIMAL_31_1_COL     DECIMAL(31, 1),
    NUMERIC_31_1_COL     NUMERIC(31, 1),
    NUM_31_1_COL         NUM(31, 1),

    DEC_31_31_COL        DEC(31, 31),
    DECIMAL_31_31_COL    DECIMAL(31, 31),
    NUMERIC_31_31_COL    NUMERIC(31, 31),
    NUM_31_31_COL        NUM(31, 31)

)