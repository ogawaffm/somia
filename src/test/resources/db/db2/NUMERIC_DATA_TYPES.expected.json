{
  "name": "NUMERIC_DATA_TYPES",
  "catalogName": "",
  "schemaName": "TEST",
  "typeName": "TABLE",
  "comment": "",
  "columns": [
    {
      "name": "SMALLINT_COL",
      "type": {
        "name": "SMALLINT",
        "typeNumber": 5,
        "className": "java.lang.Short",
        "precision": 5,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": true,
        "isCurrency": false,
        "columnDisplaySize": 6,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "INT_COL",
      "type": {
        "name": "INTEGER",
        "typeNumber": 4,
        "className": "java.lang.Integer",
        "precision": 10,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": true,
        "isCurrency": false,
        "columnDisplaySize": 11,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "INTEGER_COL",
      "type": {
        "name": "INTEGER",
        "typeNumber": 4,
        "className": "java.lang.Integer",
        "precision": 10,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": true,
        "isCurrency": false,
        "columnDisplaySize": 11,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "BIGINT_COL",
      "type": {
        "name": "BIGINT",
        "typeNumber": -5,
        "className": "java.lang.Long",
        "precision": 19,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": true,
        "isCurrency": false,
        "columnDisplaySize": 20,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "REAL_COL",
      "type": {
        "name": "REAL",
        "typeNumber": 7,
        "className": "java.lang.Float",
        "precision": 24,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 2,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": true,
        "isCurrency": false,
        "columnDisplaySize": 24,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "FLOAT_COL",
      "type": {
        "name": "DOUBLE",
        "typeNumber": 8,
        "className": "java.lang.Double",
        "precision": 53,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 2,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": true,
        "isCurrency": false,
        "columnDisplaySize": 24,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "FLOAT_1_COL",
      "type": {
        "name": "REAL",
        "typeNumber": 7,
        "className": "java.lang.Float",
        "precision": 24,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 2,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": true,
        "isCurrency": false,
        "columnDisplaySize": 24,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "FLOAT_24_COL",
      "type": {
        "name": "REAL",
        "typeNumber": 7,
        "className": "java.lang.Float",
        "precision": 24,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 2,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": true,
        "isCurrency": false,
        "columnDisplaySize": 24,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "FLOAT_25_COL",
      "type": {
        "name": "DOUBLE",
        "typeNumber": 8,
        "className": "java.lang.Double",
        "precision": 53,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 2,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": true,
        "isCurrency": false,
        "columnDisplaySize": 24,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "FLOAT_53_COL",
      "type": {
        "name": "DOUBLE",
        "typeNumber": 8,
        "className": "java.lang.Double",
        "precision": 53,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 2,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": true,
        "isCurrency": false,
        "columnDisplaySize": 24,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "DOUBLE_COL",
      "type": {
        "name": "DOUBLE",
        "typeNumber": 8,
        "className": "java.lang.Double",
        "precision": 53,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 2,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": true,
        "isCurrency": false,
        "columnDisplaySize": 24,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "DOUBLE_PRECISION_COL",
      "type": {
        "name": "DOUBLE",
        "typeNumber": 8,
        "className": "java.lang.Double",
        "precision": 53,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 2,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": true,
        "isCurrency": false,
        "columnDisplaySize": 24,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "DECFLOAT_COL",
      "type": {
        "name": "DECFLOAT",
        "typeNumber": 1111,
        "className": "java.lang.Object",
        "precision": 34,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 0,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "DECFLOAT_16_COL",
      "type": {
        "name": "DECFLOAT",
        "typeNumber": 1111,
        "className": "java.lang.Object",
        "precision": 16,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 0,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "DECFLOAT_34_COL",
      "type": {
        "name": "DECFLOAT",
        "typeNumber": 1111,
        "className": "java.lang.Object",
        "precision": 34,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 0,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "DEC_COL",
      "type": {
        "name": "DECIMAL",
        "typeNumber": 3,
        "className": "java.math.BigDecimal",
        "precision": 5,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": true,
        "isCurrency": false,
        "columnDisplaySize": 5,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "DECIMAL_COL",
      "type": {
        "name": "DECIMAL",
        "typeNumber": 3,
        "className": "java.math.BigDecimal",
        "precision": 5,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": true,
        "isCurrency": false,
        "columnDisplaySize": 5,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "NUMERIC_COL",
      "type": {
        "name": "DECIMAL",
        "typeNumber": 3,
        "className": "java.math.BigDecimal",
        "precision": 5,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": true,
        "isCurrency": false,
        "columnDisplaySize": 5,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "NUM_COL",
      "type": {
        "name": "DECIMAL",
        "typeNumber": 3,
        "className": "java.math.BigDecimal",
        "precision": 5,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": true,
        "isCurrency": false,
        "columnDisplaySize": 5,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "DEC_1_COL",
      "type": {
        "name": "DECIMAL",
        "typeNumber": 3,
        "className": "java.math.BigDecimal",
        "precision": 1,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": true,
        "isCurrency": false,
        "columnDisplaySize": 1,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "DECIMAL_1_COL",
      "type": {
        "name": "DECIMAL",
        "typeNumber": 3,
        "className": "java.math.BigDecimal",
        "precision": 1,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": true,
        "isCurrency": false,
        "columnDisplaySize": 1,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "NUMERIC_1_COL",
      "type": {
        "name": "DECIMAL",
        "typeNumber": 3,
        "className": "java.math.BigDecimal",
        "precision": 1,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": true,
        "isCurrency": false,
        "columnDisplaySize": 1,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "NUM_1_COL",
      "type": {
        "name": "DECIMAL",
        "typeNumber": 3,
        "className": "java.math.BigDecimal",
        "precision": 1,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": true,
        "isCurrency": false,
        "columnDisplaySize": 1,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "DEC_1_0_COL",
      "type": {
        "name": "DECIMAL",
        "typeNumber": 3,
        "className": "java.math.BigDecimal",
        "precision": 1,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": true,
        "isCurrency": false,
        "columnDisplaySize": 1,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "DECIMAL_1_0_COL",
      "type": {
        "name": "DECIMAL",
        "typeNumber": 3,
        "className": "java.math.BigDecimal",
        "precision": 1,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": true,
        "isCurrency": false,
        "columnDisplaySize": 1,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "NUMERIC_1_0_COL",
      "type": {
        "name": "DECIMAL",
        "typeNumber": 3,
        "className": "java.math.BigDecimal",
        "precision": 1,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": true,
        "isCurrency": false,
        "columnDisplaySize": 1,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "NUM_1_0_COL",
      "type": {
        "name": "DECIMAL",
        "typeNumber": 3,
        "className": "java.math.BigDecimal",
        "precision": 1,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": true,
        "isCurrency": false,
        "columnDisplaySize": 1,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "DEC_1_1_COL",
      "type": {
        "name": "DECIMAL",
        "typeNumber": 3,
        "className": "java.math.BigDecimal",
        "precision": 1,
        "scale": 1,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": true,
        "isCurrency": false,
        "columnDisplaySize": 1,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "DECIMAL_1_1_COL",
      "type": {
        "name": "DECIMAL",
        "typeNumber": 3,
        "className": "java.math.BigDecimal",
        "precision": 1,
        "scale": 1,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": true,
        "isCurrency": false,
        "columnDisplaySize": 1,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "NUMERIC_1_1_COL",
      "type": {
        "name": "DECIMAL",
        "typeNumber": 3,
        "className": "java.math.BigDecimal",
        "precision": 1,
        "scale": 1,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": true,
        "isCurrency": false,
        "columnDisplaySize": 1,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "NUM_1_1_COL",
      "type": {
        "name": "DECIMAL",
        "typeNumber": 3,
        "className": "java.math.BigDecimal",
        "precision": 1,
        "scale": 1,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": true,
        "isCurrency": false,
        "columnDisplaySize": 1,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "DEC_31_COL",
      "type": {
        "name": "DECIMAL",
        "typeNumber": 3,
        "className": "java.math.BigDecimal",
        "precision": 31,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": true,
        "isCurrency": false,
        "columnDisplaySize": 31,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "DECIMAL_31_COL",
      "type": {
        "name": "DECIMAL",
        "typeNumber": 3,
        "className": "java.math.BigDecimal",
        "precision": 31,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": true,
        "isCurrency": false,
        "columnDisplaySize": 31,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "NUMERIC_31_COL",
      "type": {
        "name": "DECIMAL",
        "typeNumber": 3,
        "className": "java.math.BigDecimal",
        "precision": 31,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": true,
        "isCurrency": false,
        "columnDisplaySize": 31,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "NUM_31_COL",
      "type": {
        "name": "DECIMAL",
        "typeNumber": 3,
        "className": "java.math.BigDecimal",
        "precision": 31,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": true,
        "isCurrency": false,
        "columnDisplaySize": 31,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "DEC_31_0_COL",
      "type": {
        "name": "DECIMAL",
        "typeNumber": 3,
        "className": "java.math.BigDecimal",
        "precision": 31,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": true,
        "isCurrency": false,
        "columnDisplaySize": 31,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "DECIMAL_31_0_COL",
      "type": {
        "name": "DECIMAL",
        "typeNumber": 3,
        "className": "java.math.BigDecimal",
        "precision": 31,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": true,
        "isCurrency": false,
        "columnDisplaySize": 31,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "NUMERIC_31_0_COL",
      "type": {
        "name": "DECIMAL",
        "typeNumber": 3,
        "className": "java.math.BigDecimal",
        "precision": 31,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": true,
        "isCurrency": false,
        "columnDisplaySize": 31,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "NUM_31_0_COL",
      "type": {
        "name": "DECIMAL",
        "typeNumber": 3,
        "className": "java.math.BigDecimal",
        "precision": 31,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": true,
        "isCurrency": false,
        "columnDisplaySize": 31,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "DEC_31_1_COL",
      "type": {
        "name": "DECIMAL",
        "typeNumber": 3,
        "className": "java.math.BigDecimal",
        "precision": 31,
        "scale": 1,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": true,
        "isCurrency": false,
        "columnDisplaySize": 31,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "DECIMAL_31_1_COL",
      "type": {
        "name": "DECIMAL",
        "typeNumber": 3,
        "className": "java.math.BigDecimal",
        "precision": 31,
        "scale": 1,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": true,
        "isCurrency": false,
        "columnDisplaySize": 31,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "NUMERIC_31_1_COL",
      "type": {
        "name": "DECIMAL",
        "typeNumber": 3,
        "className": "java.math.BigDecimal",
        "precision": 31,
        "scale": 1,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": true,
        "isCurrency": false,
        "columnDisplaySize": 31,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "NUM_31_1_COL",
      "type": {
        "name": "DECIMAL",
        "typeNumber": 3,
        "className": "java.math.BigDecimal",
        "precision": 31,
        "scale": 1,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": true,
        "isCurrency": false,
        "columnDisplaySize": 31,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "DEC_31_31_COL",
      "type": {
        "name": "DECIMAL",
        "typeNumber": 3,
        "className": "java.math.BigDecimal",
        "precision": 31,
        "scale": 31,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": true,
        "isCurrency": false,
        "columnDisplaySize": 31,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "DECIMAL_31_31_COL",
      "type": {
        "name": "DECIMAL",
        "typeNumber": 3,
        "className": "java.math.BigDecimal",
        "precision": 31,
        "scale": 31,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": true,
        "isCurrency": false,
        "columnDisplaySize": 31,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "NUMERIC_31_31_COL",
      "type": {
        "name": "DECIMAL",
        "typeNumber": 3,
        "className": "java.math.BigDecimal",
        "precision": 31,
        "scale": 31,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": true,
        "isCurrency": false,
        "columnDisplaySize": 31,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "NUM_31_31_COL",
      "type": {
        "name": "DECIMAL",
        "typeNumber": 3,
        "className": "java.math.BigDecimal",
        "precision": 31,
        "scale": 31,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": true,
        "isCurrency": false,
        "columnDisplaySize": 31,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    }
  ],
  "indexes": [
    {
      "name": "",
      "columns": [],
      "isUnique": false,
      "comment": "",
      "qualifierName": "",
      "isClustered": false,
      "isHashed": false,
      "isStatistic": true,
      "whereClause": "",
      "cardinality": 0,
      "pages": 0
    }
  ],
  "collation": "",
  "primaryKey": null,
  "bestRowIdentifier": null
}
